### PR TITLE
Refactor codebase to group similar functions and limit file size

### PR DESCRIPTION
**Pull request by lgaleana-llm**
Refactor the codebase to group functions by similarity into the same files. Ensure that no file exceeds 100 lines of code. The refactoring includes creating new modules for routes, OpenAI utilities, web scraping, and templates. Update imports and references accordingly. Maintain code quality and test coverage.

### Test Plan

pytest / playwright

### CODE REVIEW

**Comment by lgaleana-llm** [comment]
Total time to create PR: 365.99 seconds
Total OPenAI cost: 1.7389199999999998

File: app/routes.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+# Initialize OpenAI API key
+openai_api_key = os.getenv("OPENAI_API_KEY")
+
+app = FastAPI()
+
+# Mount the static files directory
+app.mount("/static", StaticFiles(directory="templates/static"), name="static")
+
+@app.get("/", response_class=HTMLResponse)
+def read_root(request: Request):
```
Comment (lgaleana) [inline app/routes.py:22]: Put this in `main.py`

**Review by lgaleana — APPROVED** [review]

File: app/routes.py (hunk 1 above)
Comment (lgaleana-llm) [reply app/routes.py:22]: Sure, I can move the initialization of the FastAPI app and the mounting of static files to `main.py`. I'll update the PR accordingly.

**Comment by lgaleana-llm** [comment]
Total time amending the PR: 131.72 secondsTotal OPenAI cost: 0.7408450000000001

**Review by lgaleana — APPROVED** [review]

**Pull request merged**